src/bindings.rs

```diff
@@ -622,18 +627,23 @@ impl<'a> OwnedValueRef<'a> {
     pub fn new(context: &'a ContextWrapper, value: q::JSValue) -> Self {
         Self { context, value }
     }
+    pub fn new_dup(context: &'a ContextWrapper, value: q::JSValue) -> Self {
+        let ret = Self::new(context, value);
+        unsafe{dup_value(ret.context.context, ret.value)};
+        ret
+    }
+
+    /// Get the inner JSValue without increasing ref count
+    pub(crate) fn into_inner(&self) -> &q::JSValue {
+        &self.value
+    }
 
-    /// Get the inner JSValue without freeing in drop.
-    ///
-    /// Unsafe because the caller is responsible for freeing the value.
-    //unsafe fn into_inner(mut self) -> q::JSValue {
-    //let v = self.value;
-    //self.value = q::JSValue {
-    //u: q::JSValueUnion { int32: 0 },
-    //tag: TAG_NULL,
-    //};
-    //v
-    //}
+    /// Get the inner JSValue while increasing ref count, this is handy when you pass a JSValue to a new owner like e.g. setProperty
+    #[allow(dead_code)]
+    pub(crate) fn into_inner_dup(&self) -> &q::JSValue {
```

Comment:
As above, I don't see the need for this.

If a copy is needed, `clone()` should be used.

Reply:
ok, i use methods like this as a util for when the q::JS_SomeeThing value needs a JSValue with a higher refcount, this way the enclosed JSValue in OwnedValueref is always freed as expected

e.g. getProperty requires an object without a higher refcount but when passing an object as property to setProperty you do need to increase the refcount, you could ofcourse also let the dev call DupValue... please let me know which way you want to go

Reply:
As mentioned above, this should be handled either via memory safe wrappers where possible.

EG for this case we could add a `OwnedObjectRef::set_property_owned(name: &str, value: OwnedObjectRef)` method.